medusa/show/recommendations/trakt.py

```diff
@@ -36,9 +36,9 @@ def __init__(self):
         self.cache_subfolder = __name__.split('.')[-1] if '.' in __name__ else __name__
         self.recommender = "Trakt Popular"
         self.default_img_src = 'trakt-default.png'
-        self.anidb = Anidb(cache_dir=app.CACHE_DIR)
         self.tvdb_api_v2 = indexerApi(INDEXER_TVDBV2).indexer()
 
+    @recommended_series_cache.cache_on_arguments()
```

Comment:
The argument here is a show object. You need to provide an "object to key" function.

Reply:
Maybe your right. But it seems to work fine. Maybe it pickling the object?

Reply:
It works because it's the same object instance. If the cache is persistent and you restart Medusa, the show object will be a different instance

Reply:
No it still used the correct cached key

Reply:
Give it a try, you'll see

Reply:
This is a key that has been generated:
```
medusa.show.recommendations.imdb:_create_recommended_show|{u'rating': 8.8, u'votes': 50663, u'name': u'This Is Us', u'imdb_url': u'http://www.imdb.com/title/tt5555260/', u'outline': u'A heartwarming story of the struggles of the most unique set of triplets, and the wonderful parents who raised them.', u'image_url_large': u'https://images-na.ssl-images-amazon.com/images/M/MV5BOWU3M2I0YWItNWIzNC00MTBmLTkzMDgtMTdmOGMzOTI2YmVjXkEyXkFqcGdeQXVyNjMxNzcwOTI@._V1_.jpg', u'image_url': u'https://images-na.ssl-images-amazon.com/images/M/MV5BOWU3M2I0YWItNWIzNC00MTBmLTkzMDgtMTdmOGMzOTI2YmVjXkEyXkFqcGdeQXVyNjMxNzcwOTI@.__SY600_AL_.jpg', u'year': 2016, u'imdb_tt': u'tt5555260', u'image_path': u'images/imdb_popular/MV5BOWU3M2I0YWItNWIzNC00MTBmLTkzMDgtMTdmOGMzOTI2YmVjXkEyXkFqcGdeQXVyNjMxNzcwOTI@._V1_.jpg'}
```

It could still be better to calculate the key our selves. But that would be to not make it refresh, every time the rating or number of votes changes?

Reply:
I see the confusion now. I named it `show_obj` while in fact it's a dict. That's why dogpile can handle it. It would be really different if a real object was passed.

Reply:
I'm getting the hang of it. This dogpile hocus pocus is really awesome. I might even just use it to create an anidb indexer. As we need heavy caching for that.